lnwallet/update_log.go

```diff
@@ -27,16 +29,16 @@ type updateLog struct {
 
 	// List is the updatelog itself, we embed this value so updateLog has
 	// access to all the method of a list.List.
-	*list.List
+	*fn.List[*PaymentDescriptor]
 
 	// updateIndex maps a `logIndex` to a particular update entry. It
 	// deals with the four update types:
 	//   `Fail|MalformedFail|Settle|FeeUpdate`
-	updateIndex map[uint64]*list.Element
+	updateIndex map[uint64]*fn.Node[*PaymentDescriptor]
```

Comment:
hmm why pointers?

Reply:
We currently mutate the values in the list. This was the smallest change I could make. Future commits in the series are planning to remove `PaymentDescriptor` from this list altogether. One thing at a time.